tests/playwright/src/model/workbench/navigation.ts

```diff
@@ -70,8 +70,11 @@ export class NavigationBar {
   }
 
   async openSettings(): Promise<SettingsBar> {
-    await this.settingsLink.waitFor({ state: 'visible', timeout: 3000 });
-    await this.settingsLink.click({ timeout: 5000 });
-    return new SettingsBar(this.page);
+    const settingsBar = new SettingsBar(this.page);
+    if (!(await settingsBar.settingsNavBar.isVisible())) {
+      await this.settingsLink.waitFor({ state: 'visible', timeout: 3000 });
```

Comment:
We shouldn't use .waitFor() in such cases, if something is needed to be awaited for visibility it's recommended to use `await except(locator).toBeVisible()` however in this case the since the item is clickable it might be better to use .toBeEnabled().

Reply:
Problem is the behavior of the link. If it is visible, do not click it, since it will disapper. So the visibility check is fine. We do not want to click on the element itself, but if it is visible, we do not want to click on other one. 
Is `waitFor` use in `locator.isVisible()` method?

Reply:
Ah, the waitFor that was already part of code. got it. Will change.